tools/run_cargo_generate-lockfile.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Runs `cargo generate-lockfile` (which generates and/or updates the Cargo.lock
+# file for a crate) on every subdirectory from . that has a Cargo.toml file.
+#
+# Author: Brad Campbell <bradjc5@gmail.com>
+
+FAIL=0
+
+set -e
+
+# Verify that we're running in the base directory
+if [ ! -x tools/run_cargo_generate-lockfile.sh ]; then
+	echo ERROR: $0 must be run from the tock repository root.
+	echo ""
+	exit 1
+fi
+
+for f in $(find . | grep Cargo.toml); do
+	pushd $(dirname $f) > /dev/null
+	cargo generate-lockfile || let FAIL=FAIL+1
+	popd > /dev/null
+done
```

Comment:
Something around here needs to check the result of `FAIL`, i.e.

```
if [[ $FAIL -ne 0 ]]; then
	echo
	echo "$(tput bold)Some lockfile generation failed$(tput sgr0)"
	echo "See above for details"
	exit $FAIL
fi
```

Alternatively, remove the `|| let FAIL=FAIL+1` bit and just error out the first time any one fails. Right now though, this script will hide errors and always claim success